src/osd/OSD.cc

```diff
@@ -8184,7 +8205,15 @@ void OSD::handle_osd_map(MOSDMap *m)
   {
     bufferlist bl;
     ::encode(pg_num_history, bl);
-    t.write(coll_t::meta(), make_pg_num_history_oid(), 0, bl.length(), bl);
+    auto oid = make_pg_num_history_oid();
+    t.truncate(coll_t::meta(), oid, 0); // we don't need bytes left if new data
```

Comment:
Doing truncate is an excellent idea.
It will empty object content, and will force re-allocation of space for `osd_superblock`.
We no longer will have the case that `osd_superblock` is parked forever as first data on the device, always updated in-place by deferred writes.
And I think an info to this effect should be added to code comments.

Reply:
done.